data_processing/services/graph_service.py

```diff
@@ -0,0 +1,66 @@
+import os
+import click
+import pandas as pd
+from pyspark.sql import functions as F
+
+from data_processing.common.GraphEnum import GraphEnum
+from data_processing.common.Neo4jConnection import Neo4jConnection
+from data_processing.common.custom_logger import init_logger
+from data_processing.common.sparksession import SparkConfig
+from data_processing.common.config import ConfigSet
+import data_processing.common.constants as const
+
+@click.command()
+@click.option('-p', '--project', help="project name", required=True)
+@click.option('-t', '--table', help="table name", required=True)
+@click.option('-f', '--config_file', default = 'config.yaml', required=True,
+              help="path to config file containing application configuration. See config.yaml.template")
+def update_graph(project, table, config_file):
+
+	logger = init_logger('update_graph.log')
+
+	# Set up : Neo4j connection and Spark session
+	logger.info("Setting up graph connection and spark session")
+	cfg = ConfigSet(name=const.APP_CFG, config_file=config_file)
+	spark = SparkConfig().spark_session(config_name=const.APP_CFG, app_name="update-graph")
+
+	conn = Neo4jConnection(uri=cfg.get_value(name=const.APP_CFG, jsonpath='GRAPH_URI'),
+		user=cfg.get_value(name=const.APP_CFG, jsonpath='GRAPH_USER'),
+		pwd=cfg.get_value(name=const.APP_CFG, jsonpath='GRAPH_PW'))
+
+	# get project / table path
+	if "MIND_ROOT_DIR" in os.environ:
+		base_path = os.environ["MIND_ROOT_DIR"]
+	else:
+		base_path = "/gpfs/mskmindhdp_emc/data/"
```

Comment:
I thought about this too! I hesitated because of 1) our current (now old) convention and 2) MIND_ROOT_DIR fitting more into DATA_CFG. But for simplicity, i think we can have it in the app config.

Reply:
Consensus after offline discussion is to not extend the data ingestion template to include kvps that are needed for other functions like graph updates or scan table generation. To limit data ingestion template merely to generating the proxy table which is the result of data ingestion. To follow a pattern of making a yaml config file initially which is very specific to the table and data type and where the template can be maintained not at the root dir of the repo but in the appropriate package. And then try to generalize to all modalities and apply schema and validation if at all it makes sense to do so in the future. But at minimum start with a simple yaml config.